Sync `tools` from WPT upstream

TingPing opened the pull request:
#### 3a03865ff84d2be82f5b0e62c3a00e892a038388
<pre>
Sync `tools` from WPT upstream
<a href="https://bugs.webkit.org/show_bug.cgi?id=275992">https://bugs.webkit.org/show_bug.cgi?id=275992</a>

Reviewed by Sam Sneddon.

Upstream commit: <a href="https://github.com/web-platform-tests/wpt/commit/e4cf05764e3b0239593e2ccb4761d96a68a0efbc">https://github.com/web-platform-tests/wpt/commit/e4cf05764e3b0239593e2ccb4761d96a68a0efbc</a>

This fixes various issues running on Python 3.12.

* LayoutTests/imported/w3c/web-platform-tests/tools/:

Canonical link: <a href="https://commits.webkit.org/280476@main">https://commits.webkit.org/280476@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/d97f44dc6aaa220c7bf8cde9e3716ac034604ac8

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [❌ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/56698 "219 style errors") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/36025 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/9171 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/60312 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/59/builds/7139 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/58825 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/43648 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/7330 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/45929 "Passed tests") | [✅ 🧪 wincairo-tests](https://ews-build.webkit.org/#/builders/60/builds/5007 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/58728 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/33864 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/48936 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/26788 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/30643 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/6268 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/6142 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/52601 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/6539 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/61995 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/87/builds/610 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/62/builds/6639 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/53189 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/86/builds/612 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/48994 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/53207 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/12551 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/88/builds/526 "Passed tests") | | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/31853 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/32939 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/34023 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/32685 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

gsnedders reviewed (approved):
I tried doing this recently and I'm unsure why I didn't land this?

Oh, right, I think it was https://github.com/web-platform-tests/wpt/issues/46040.

Well, I fixed that a while ago, so r+ from me here provided it passes EWS, which it has!

TingPing committed:
Sync `tools` from WPT upstream
https://bugs.webkit.org/show_bug.cgi?id=275992

Reviewed by Sam Sneddon.

Upstream commit: https://github.com/web-platform-tests/wpt/commit/e4cf05764e3b0239593e2ccb4761d96a68a0efbc

This fixes various issues running on Python 3.12.

* LayoutTests/imported/w3c/web-platform-tests/tools/:

Canonical link: https://commits.webkit.org/280476@main
